config/config.go

```diff
@@ -204,7 +204,7 @@ type Config struct {
 	RuleFiles      []string        `yaml:"rule_files,omitempty"`
 	ScrapeConfigs  []*ScrapeConfig `yaml:"scrape_configs,omitempty"`
 
-	RemoteWriteConfig RemoteWriteConfig `yaml:"remote_write,omitempty"`
```

Comment:
What I mean is

```yaml
# ...
remote:
  write:
    some_param: 123
    more_stuff: "test
  read:
    totally_differant params: 123

# ...
```

Of course they'll be vastly different. But what's the point in having separate fields with the same prefix. That's literally flatting hierarchy into opaque strings.